### DISCUSSION TITLE
Error messages when installing charm-gems

### DISCUSSION THREAD

**Discussion by Akira-Kita** [simnibs/simnibs#272] [6. Getting started] [answered]
Hello experts.

I am an beginner of SimNibs.
I try to install the charm-gems to my linux woring in my Windows11.
I used UBUNTU22.04.3 in wsl in Windows.
I already succeeded in building ITK.
Then I ordered the command 'ITK_DIR=ITK-install python3 setup.py install' in my terminal.
But the following error messages came.
My python3 version is 59.6.0.

'In file included from /home/akira/charm-gems-master/gems/itkMGHImageIO.h:11,
                 from /home/akira/charm-gems-master/gems/itkMGHImageIO.cxx:4:
/home/akira/charm-gems-master/ITK-install/include/ITK-5.4/itkExceptionObject.h:19:4: error: #error "Do not include itkExceptionObject.h directly,  include itkMacro.h instead."
   19 | #  error "Do not include itkExceptionObject.h directly,  include itkMacro.h instead."
      |    ^~~~~
make[2]: *** [gems/CMakeFiles/kvlGEMSCommon.dir/build.make:90: gems/CMakeFiles/kvlGEMSCommon.dir/itkMGHImageIO.cxx.o] Error 1
make[2]: Leaving directory '/tmp/tmpoqeqh_z7'
make[1]: *** [CMakeFiles/Makefile2:134: gems/CMakeFiles/kvlGEMSCommon.dir/all] Error 2
make[1]: Leaving directory '/tmp/tmpoqeqh_z7'
make: *** [Makefile:91: all] Error 2
make: Leaving directory '/tmp/tmpoqeqh_z7''

I attached the error message file to this thread.

Any comment is appreciated.

Thank you.

Best
Akira
[s-14.docx](https://github.com/simnibs/simnibs/files/13903141/s-14.docx)
[error.txt](https://github.com/simnibs/simnibs/files/13903143/error.txt)

**Comment by oulap** [comment, marked as answer]
If you want to install SimNIBS (which includes charm-gems) you don't need to build it from scratch. Just use the linux installer, which you can download from here:

https://github.com/simnibs/simnibs/releases/tag/dev

If you specifically only want to install charm-gems, you can pip install it from here:

https://pypi.org/project/charm-gems/

or preferably here:

https://pypi.org/project/samseg/

**Reply by Akira-Kita** [reply]
Thank you for your helpful comment.
According to the comments, I tried to install simnibs using the simnibs_installer.
And I ordered the folloing command, 'simnibs_installer/install
' according to the guide (https://simnibs.github.io/simnibs/build/html/installation/simnibs_installer.html).

However, the following error came again.
'simnibs_installer/install
Installing SimNIBS 4.1.0a1.dev23+ge15afab to /home/akira/SimNIBS-4.1
Found another SimNIBS installation in /home/akira/SimNIBS-4.1. Remove it? [y/N] :y
Fontconfig error: Cannot load default config file: No such file: (null)
libGL error: MESA-LOADER: failed to open swrast: /home/akira/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages/scipy/spatial/../../../../libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /lib/x86_64-linux-gnu/libLLVM-15.so.1) (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)
libGL error: failed to load driver: swrast
PyQt workaround: QT_QPA_PLATFORM_PLUGIN_PATH set to /home/akira/SimNIBS-4.1/simnibs_env/plugins/platforms
Traceback (most recent call last):
  File "/home/akira/SimNIBS-4.1/simnibs_env/bin/postinstall_simnibs", line 8, in <module>
    sys.exit(main())
  File "/home/akira/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages/simnibs/cli/postinstall_simnibs.py", line 1033, in main
    start_gui(
  File "/home/akira/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages/simnibs/cli/postinstall_simnibs.py", line 858, in start_gui
    raise Exception('Post-installation cancelled by user')
Exception: Post-installation cancelled by user'

I also attached the file of the error message in this thread.

Any comment is appreciated. Thank you again for your kind helps!
[error_simnibs_installerinstall.txt](https://github.com/simnibs/simnibs/files/13917538/error_simnibs_installerinstall.txt)


Best
Akira

**Comment by jduemose** [comment]
Hi Akira, it seems you might have to preload your system's libstd++.so. Have a look at these discussions and see if any of the suggestions help #87 and #153.
Best,
Jesper

**Reply by Akira-Kita** [reply]
Thank you for your help. I tried #153 and solved installing simnibs. Thanks!!

**Reply by Akira-Kita** [reply]
When installing pynibs in UBUNTU in Windows11, the error messages as follows came.
Does anyone know how to solve this issue?
Any comment is appreciated.

Best 
Akira 

Installed /home/akira/pynibs
Processing dependencies for pynibs==0.2023.8
Searching for tables
Reading https://pypi.org/simple/tables/
Download error on https://pypi.org/simple/tables/: [Errno 104] Connection reset by peer -- Some packages may not be found!
Couldn't find index page for 'tables' (maybe misspelled?)
Scanning index of all packages (this may take a while)
Reading https://pypi.org/simple/
No local packages or working download links found for tables
error: Could not find suitable distribution for Requirement.parse('tables')

**Reply by TorgeW** [reply]
For pynibs questions, please refer to the pynibs support.

**Reply by Akira-Kita** [reply]
Thank you for your helpful comment.
I have solved by installing pynibs.
Thanks!
